Netgroup conf

boandersson opened the pull request:
Add configurable parameter for netgroup-mode in vas.conf. Specifically required for turning off vasd netgroup handling for a site that doesn't use netgroups (join fails unless turning it off.)
Code tested in our environment.

jonathanbower commented:
Can we please have this merged?

Phil-Friderici commented on README.md:
```diff
@@ -373,6 +373,12 @@ Path for script to set value of delusercheck-script in [vasd] section of vas.con
+vas_conf_vasd_netgroup_mode
+---------------------------
+String to be used to set value of netgroup-mode in the [vasd] section of vas.conf. See VAS.CONF(5) for more info.
+
```
Please add a hint which values are accepted.

Phil-Friderici commented on spec/classes/init_spec.rb:
```diff
@@ -339,6 +341,29 @@
+        end
+      end
+    end
+
```
Please use the 'variable type and content validations' section at the bottom for the type and content validations.

With that in place, atomic tests of your functionality are good enough. Take the tests for $join_domain_controllers as example.

boandersson commented:
Thanks for looking into this. Agree with the needless implementation of the validation tests. Let me know what you think about the updates.

Phil-Friderici commented:
Would like to see specific 'atomic' tests of the new functionality. Something like:

```ruby
context 'with vas_conf_vasd_netgroup_mode set to valid string NSS' do
  let(:params) { { :vas_conf_vasd_netgroup_mode => 'NSS', } }
  it { should contain_file('vas_config').with_content(/^netgroup-mode = NSS$/) }
end

context 'with vas_conf_vasd_netgroup_mode set to valid string UNSET' do
  let(:params) { { :vas_conf_vasd_netgroup_mode => 'UNSET', } }
  it { should contain_file('vas_config').without_content(/^netgroup-mode/) }
end
```
(good chance that the above example could work out of the box)

I know that these tests are implicit included inside the tests for defaults only and with parms on RH6, just like to see them tested explicitly on their own.

After this is done, please squash your commits into one so that I can merge them then :)

boandersson commented:
Thanks! I realized what you meant by "atomic" test only after committing when digging through previous changes for making travis happy :)
Added your suggestions, squashed the commits and rebased onto master.

Phil-Friderici commented:
Travis seems stuck. Have re-triggered it one more time in https://github.com/Ericsson/puppet-module-vas/pull/114.
Keeping fingers crossed to have green results tomorrow morning.

Phil-Friderici commented:
Spec tests were run successfully in https://github.com/Phil-Friderici/puppet-module-vas/pull/1 as workaround for Travis CI not getting triggerd in this repository.

Phil-Friderici commented:
@boandersson Thanks ! Released in v0.22.0
